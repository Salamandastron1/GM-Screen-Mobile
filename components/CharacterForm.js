import React, { Component } from 'react';
import { Alert } from 'react-native';
import PropTypes from 'prop-types';
import MainContain from './MainContain';
import CardSection from './CardSection';
import Title from './Title';
import Input from './Input';
import Button from './Button';

class CharacterForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
    };

    this.handleInput = this.handleInput.bind(this);
    this.determineRoute = this.determineRoute.bind(this);
  }

  handleInput({ nativeEvent }) {
    this.setState({
      name: nativeEvent.text,
    });
  }

  determineRoute(name) {
    if (name.length === 0) {
      return Alert.alert('Please give your character a name');
    }
    const { navigation } = this.props;
    const characterId = navigation.getParam('id');
    const url = `/api/v1/characters/${characterId}`;
    const data = { name };

    fetch(url, {
      method: 'PUT',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json',
      },
    });
    navigation.navigate('PlayerScreen', { name });

    return undefined;
  }

  render() {
    const { name } = this.state;

    return (
      <MainContain>
        <CardSection>
          <Input
            value={name}
            onChange={this.handleInput}
            name="name"
            length={20}
            placeholder="Character Name"
          />
        </CardSection>
        <CardSection>
          <Button
            onPress={() => this.determineRoute(name)}
          >
            Submit
          </Button>
        </CardSection>
        <Title>
          Create Your Character
        </Title>
      </MainContain>
    );
  }
}

CharacterForm.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default CharacterForm;
