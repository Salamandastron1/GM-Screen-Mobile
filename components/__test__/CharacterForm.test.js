/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import CharacterForm from '../CharacterForm';

it('should match the shapshot', () => {
  const tree = shallow(<CharacterForm />);
  expect(tree).toMatchSnapshot();
});