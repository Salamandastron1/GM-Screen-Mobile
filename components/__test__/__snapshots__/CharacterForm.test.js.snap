// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the shapshot 1`] = `
<MainContain>
  <CardSection>
    <Input
      length={20}
      name="name"
      onChange={[Function]}
      placeholder="Character Name"
      value=""
    />
  </CardSection>
  <CardSection>
    <Button
      onPress={[Function]}
    >
      Submit
    </Button>
  </CardSection>
  <Title>
    Create Your Character
  </Title>
</MainContain>
`;
